server/datastore/mysql/hosts.go

```diff
@@ -21,66 +21,83 @@ import (
 	"github.com/jmoiron/sqlx"
 )
 
-var hostSearchColumns = []string{"hostname", "uuid", "hardware_serial", "primary_ip"}
+var hostSearchColumns = []string{"hostname", "computer_name", "uuid", "hardware_serial", "primary_ip"}
 
 // NewHost creates a new host on the datastore.
 //
 // Currently only used for testing.
 func (ds *Datastore) NewHost(ctx context.Context, host *fleet.Host) (*fleet.Host, error) {
-	sqlStatement := `
-	INSERT INTO hosts (
-		osquery_host_id,
-		detail_updated_at,
-		label_updated_at,
-		policy_updated_at,
-		node_key,
-		hostname,
-		uuid,
-		platform,
-		osquery_version,
-		os_version,
-		uptime,
-		memory,
-		team_id,
-		distributed_interval,
-		logger_tls_period,
-		config_tls_refresh,
-		refetch_requested
-	)
-	VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
-	`
-	result, err := ds.writer.ExecContext(
-		ctx,
-		sqlStatement,
-		host.OsqueryHostID,
-		host.DetailUpdatedAt,
-		host.LabelUpdatedAt,
-		host.PolicyUpdatedAt,
-		host.NodeKey,
-		host.Hostname,
-		host.UUID,
-		host.Platform,
-		host.OsqueryVersion,
-		host.OSVersion,
-		host.Uptime,
-		host.Memory,
-		host.TeamID,
-		host.DistributedInterval,
-		host.LoggerTLSPeriod,
-		host.ConfigTLSRefresh,
-		host.RefetchRequested,
-	)
-	if err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "new host")
-	}
-	id, _ := result.LastInsertId()
-	host.ID = uint(id)
+	err := ds.withTx(ctx, func(tx sqlx.ExtContext) error {
```

Comment:
Ah, I get what you're going for here (and elsewhere in this file) - I did the same thing when I started! Adding transactions where operations looked like they should be atomic. It turned out we removed most of them later on due to lock contentions/slow under load (@chiiph might remember better the exact issues). We really should document that, as I think it's a very natural reaction when reading such code to bring back transactions.

We don't really suffer from a lack of atomicity regarding host data as it is constantly being reported by hosts checking in, and it is eventually consistent in nature.

Reply:
Haven't read the PR, but I would say: let's throw this into loadtest and see how it behaves compared to main. We can make the decision with data. As long as the tx is small and fast, it's all good.

Reply:
+1 who can show me how to run the loudtest?

A quick Benchmark that uses 20 coroutines to alternately create hosts and list the first 20 sorted by display_name doesn't show a significant change.

```
~/go/bin/benchstat /tmp/withoutTx /tmp/withTx                                                         
name                  old time/op  new time/op  delta
Datastore_NewHost-10   156ms ± 4%   160ms ± 7%   ~     (p=0.113 n=10+9)
```